mk/xe-guest-utilities.spec.in

```diff
@@ -75,8 +79,16 @@ cp %{SOURCE8} %{buildroot}/usr/share/doc/%{name}-xenstore-%{version}/
 rm -rf %{buildroot}
 
 %post
-/sbin/chkconfig --add xe-linux-distribution >/dev/null
-[ -n "${EXTERNAL_P2V}" ] || service xe-linux-distribution start >/dev/null 2>&1
+#!/bin/bash
+if command -v systemctl >/dev/null 2>&1; then
+    xe_install_path=/usr/sbin
+    sed -i "s#/usr/share/oem/xs#$xe_install_path#g" /usr/lib/systemd/system/xe-linux-distribution.service
+    systemctl enable /usr/lib/systemd/system/xe-linux-distribution.service >/dev/null 2>&1
+    systemctl start xe-linux-distribution.service >/dev/null 2>&1
+else
+    /sbin/chkconfig --add xe-linux-distribution >/dev/null
```

Comment:
do we still have any distribution need to support that does not have systemd?
if no, clean the chkconfig completely.

Reply:
RHEL 6 still not support systemd

Reply:
We already tried using a single RPM for both chkconfig and systemd at XCP-ng, and found out that it wouldn't work in some distros. I don't remember the exact details, but it had to do with code in systemd which may detect the presence of the systemV service and not start the service correctly. I think I remember a version of SLES in this situation, but I don't remember exactly which.

That's why we created two different RPMs instead of one, and it's up to `install.sh` to install the right one.

It is detailed in this PR: https://github.com/xcp-ng-rpms/xcp-ng-pv-tools/pull/4/files

We also had to handle the transition from chkconfig to systemd. In theory systemd handles it, but not all version were found to do it perfectly. That's why you'll find extra steps in %post: https://github.com/xcp-ng-rpms/xcp-ng-pv-tools/pull/4/files#diff-71a4c36ceca2d58448dd9abf9bc21c96f5dcc95142ff1f46fae70dcd711ca3f8R53-R75

Reply:
Update: I now remember the issue with providing both service files. systemd detects the chkconfig service, attempts to start it with its SysV compatibility layer, deferring to chkconfig... And then it fails when chkconfig is not present on the system, as is the case in some recent distros.

Reply:
Our RPMs were tested with:

```
            "centos6-32-PVINPVH-created_8.2-zstd.xva",
            "centos6-32-hvm-created_8.2-zstd.xva",
            "centos6-64-PV-created_8.0-zstd.xva",
            "centos7-64-uefi-created_8.1-zstd.xva",
            "centos8-stream-64-gui-created_8.0-zstd.xva",
            "centos-stream-9-04082022-uefi-created_8.2-zstd.xva",
            "almalinux-8.5-uefi-created_8.2-zstd.xva",
            "almalinux-9-05082022-uefi-created_8.2-zstd.xva",
            "openSUSE-Leap-15-4-uefi-created_8.2-zstd.xva", 
```

Reply:
The test cases already been tested on our supported guests fully. Would you give an example that not support?

Reply:
I already tried to detail the cases where you may have issues:
* On distros which don't provide chkconfig, not even as a compatibility layer, when systemd may attempt to use it due to the presence of the xe-linux-distribution init script on the system. It may depend on the version of systemd, as this looks like a systemd bug to me (it shouldn't try the compat layer when chkconfig is absent). I don't remember the exact distros which caused this issue, but we preferred to split the RPM into two different ones for a more resilient installation on many distros. 
* When updating from the previous version of the guest tools, the systemd service may not start immediately because it thinks the service is still active, as explained in the spec file comments. It's not a blocker issue, but it means the tools may not be running after an update unless the runner restarts them manually or reboots. IIRC this happened on various distros tested, but if you already tested the transition (not just installing the tools on a fresh distro), maybe I'm wrong...

Reply:
Basically, I'm offering an example of what I think is a more robust way to handle the transition and the diversity of distros, maybe not just the ones you officially support now, but also the ones you may support in the future (only recent distros have systemd without the chkconfig compatibility layer).

Reply:
Maybe first found out the not supported one in the memory to see what happened. We already have test cases to cover any distribution we want to support, if any error been found, we will quickly found it.

Reply:
Thanks for kindly remained us for this, but after tested we do not find the same problem as you mentioned. 
by the way, as your solution, split the spec into two file to manage RPM is not a good way, it will lead you into the trouble of redundancy. and may course a lot of problem in future. I believe there must be a better approach worth considering.